go/parquet/compress/compress.go

```diff
@@ -49,8 +49,9 @@ var Codecs = struct {
 	Brotli Compression
 	// LZ4 unsupported in this library due to problematic issues between the Hadoop LZ4 spec vs regular lz4
 	// see: http://mail-archives.apache.org/mod_mbox/arrow-dev/202007.mbox/%3CCAAri41v24xuA8MGHLDvgSnE+7AAgOhiEukemW_oPNHMvfMmrWw@mail.gmail.com%3E
-	Lz4  Compression
-	Zstd Compression
+	Lz4    Compression
```

Comment:
Nit: maybe we can rename this to LZ4_HADOOP to prevent from misunderstanding

Reply:
The names here match the parquet spec names exactly, but I agree it would be helpful to limit confusion about this. I just pushed a change that returns a descriptive error when one of the unsupported codecs is used. Let me know what you think.

Reply:
Both is ok for me, we can draft a separate pr for that?

Reply:
Ok sure. I just reverted that commit, will add it back in a follow-up PR.